### PR TITLE
feat(frontend): add indicator when required field are missing

### PR DESCRIPTION
**Pull request by fatonramadani**
Screenshot 2024-06-19 at 15 20 55

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 55d2aad6e89baffeee06541b6263c7b375ba1273  | 
|--------|--------|

### Summary:
Added indicators for missing required fields in the frontend flow editor and updated various components to manage and display the state of required inputs.

**Key points**:
- **Added**: Indicator for missing required fields in `frontend/src/lib/components/ArgInput.svelte`.
- **Updated**: `frontend/src/lib/components/Dev.svelte` to include `flowInputsStore` in `FlowEditorContext`.
- **Updated**: `frontend/src/lib/components/FlowBuilder.svelte` to include `flowInputsStore` in `FlowEditorContext`.
- **Updated**: `frontend/src/lib/components/InputTransformForm.svelte` to dispatch `change` event on input change.
- **Updated**: `frontend/src/lib/components/InputTransformSchemaForm.svelte` to dispatch `changeArg` event on input change.
- **Updated**: `frontend/src/lib/components/flows/content/FlowEditorPanel.svelte` to initialize `requiredInputsFilled` state.
- **Updated**: `frontend/src/lib/components/flows/content/FlowModuleComponent.svelte` to set `requiredInputsFilled` state on input change.
- **Updated**: `frontend/src/lib/components/flows/content/FlowModuleWrapper.svelte` to initialize `requiredInputsFilled` state.
- **Updated**: `frontend/src/lib/components/flows/flowStateUtils.ts` to include functions for managing `requiredInputsFilled` state.
- **Updated**: `frontend/src/lib/components/flows/map/FlowModuleSchemaItem.svelte` to display indicator for missing required fields.
- **Updated**: `frontend/src/lib/components/flows/map/FlowModuleSchemaMap.svelte` to handle `flowInputsStore`.
- **Updated**: `frontend/src/lib/components/flows/types.ts` to include `FlowInput` type and `flowInputsStore` in `FlowEditorContext`.
- **Updated**: `frontend/src/lib/components/flows/utils.ts` to include utility functions for `requiredInputsFilled` state.
- **Updated**: `frontend/src/lib/components/graph/FlowGraph.svelte` to handle `flowInputsStore`.
- **Updated**: `frontend/src/routes/flows/dev/+page.svelte` to include `flowInputsStore` in `FlowEditorContext`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: frontend/src/lib/components/graph/util.ts [hunk 1]
```diff
@@ -35,51 +33,3 @@ export function getStateColor(state: FlowStatusModule['type'] | undefined): stri
 			return isDark ? '#2e3440' : '#fff'
 	}
 }
-
```
Comment (fatonramadani) [inline frontend/src/lib/components/graph/util.ts:38]: Dead code

File: frontend/src/lib/components/ArgInput.svelte [hunk 2]
```diff
@@ -233,6 +233,12 @@
 	let itemsLimit = 50
 
 	$: validateInput(pattern, value, required)
+
+	let oldValue = value
+	$: if (JSON.stringify(oldValue) !== JSON.stringify(value)) {
```
Comment (rubenfiszel) [inline frontend/src/lib/components/ArgInput.svelte]: deepEqual, also no inlined reactivity please, hard to read

Comment (rubenfiszel) [reply frontend/src/lib/components/ArgInput.svelte]: also had a prop that this should fire events on change, most ArgInput's dont need that kind of reactivity and ArgInput is used everywhere

Comment (rubenfiszel) [reply frontend/src/lib/components/ArgInput.svelte]: Also add debounce here, no need to fire events on every key stroke

**Pull request merged**